### PR TITLE
fix(skia): Correctly handle SolidColorBrush opacity in inlines

### LINKED ISSUE #11322: Skia doesn't respect Opacity of SolidColorBrush when rendering Text in TextBlock
### Current behavior

Define a SolidColorBrush with Opacity less than 1 eg
`<SolidColorBrush x:Key="MyBrush" Color="Red" Opacity="0.7"/>
`
Use the SolidColorBrush in a TextBlock
`<TextBlock Text="Text should be faded" Foreground="{StaticResource MyBrush}" />
`
The text renders at full (100%) opacity

### Expected behavior

Text should be rendered at defined opacity (70% in the example given)

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

Workaround is to put the TextBlock inside a Border and set the Opacity on the Border instead

<Border Opacity="0.7">
    <TextBlock Text="Faded text"/>
</Border>

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #11322

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs [hunk 1]
```diff
@@ -207,5 +211,25 @@ public async Task When_FontFamily_In_AndroidAsset()
...
+				Foreground = new SolidColorBrush(Colors.Red) { Opacity = 0.5 },
+			};
+
+			WindowHelper.WindowContent = SUT;
+			await WindowHelper.WaitForIdle();
+
+			var renderer = new RenderTargetBitmap();
+			await renderer.RenderAsync(SUT);
+			var bitmap = await RawBitmap.From(renderer, SUT);
+			ImageAssert.HasColorInRectangle(bitmap, new System.Drawing.Rectangle(0, 0, bitmap.Width, bitmap.Height), Color.FromArgb(127, 127, 0, 0));
```
Comment (MartinZikmund) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs:232]: Shouldn't the red channel be 255? Probably I am just missing something

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs [hunk 2]
```diff
@@ -212,6 +212,7 @@ public async Task When_FontFamily_In_AndroidAsset()
 			Assert.AreNotEqual(originalSize, SUT.DesiredSize);
 		}
 
+#if !__WASN__ // RenderTargetBitmap not yet available on Wasm.
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs:215]: Let's use IsTypeSupported instead

**Comment by Youssef1313** [comment]
I don't find any existing test that uses RenderTargetBitmap running on macOS. I'll skip for now on macOS.

**Pull request merged**